org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/extensions/XMLExtensionsRegistry.java

```diff
@@ -110,7 +110,10 @@ public void initializeParams(InitializeParams params) {
 	public void doSave(ISaveContext saveContext) {
 		if (initialized) {
 			extensions.stream().forEach(extension -> extension.doSave(saveContext));
-		} else {
+		} else if (this.initialSaveContext == null || (saveContext != null && saveContext.getSettings() != null)) {

```

Comment:
I think you should write the condition like this:

`} else if (this.initialSaveContext == null || (saveContext != null && saveContext.getType() == SaveContextType.SETTINGS)) {`

Reply:
Got it.